src/core/net/dhcp6_server.cpp

```diff
@@ -64,7 +64,7 @@ otError Dhcp6Server::UpdateService(void)
     Lowpan::Context                 lowpanContext;
 
     // remove dhcp agent aloc and prefix delegation
-    for (int i = 0; i < OPENTHREAD_CONFIG_DHCP6_SERVER_NUM_PREFIXES; i++)
+    for (uint8_t i = 0; i < OT_ARRAY_LENGTH(mPrefixAgents); i++)
```

Comment:
Size_t instead of uint8 to avoid conversation

Reply:
Pushed the second commit to unify the type of local variables when `OT_ARRAY_LENGTH` is used. Please review.